internal/tools/changes/release.go

```diff
@@ -75,19 +85,103 @@ func (r *Release) RenderChangelogForModule(module, headerPrefix string) (string,
 	return buff.String(), nil
 }
 
+const rootChangelogTemplateContents = `# Release {{.ID}}
+{{with .AnnouncementsSection}}## Announcements
+{{range .}}{{.}}
+{{end -}}
+{{end}}{{with .ServiceSection}}## Service Client Highlights
+{{range .}}{{.}}
+{{end -}}
+{{end}}{{with .CoreSection}}## Core SDK Highlights
+{{range .}}{{.}}
+{{end -}}
+{{end}}`
+
+// RenderChangelog generates a top level CHANGELOG.md for the Release r.
+func (r *Release) RenderChangelog() (string, error) {
+	buff := new(bytes.Buffer)
+
+	err := rootChangelogTemplate.Execute(buff, r)
+	if err != nil {
+		return "", err
+	}
+
+	return buff.String(), nil
+}
+
 // AffectedModules returns a sorted list of all modules affected by this Release. A module is considered affected if
 // it is the Module of one or more Changes in the Release.
 func (r *Release) AffectedModules() []string {
-	var modules []string
-	seen := make(map[string]struct{})
+	return AffectedModules(r.Changes)
+}
+
+// wildcards returns a sorted list of wildcards Changes whose Module begin with the given prefix.
+func (r *Release) wildcards(prefix string) []Change {
+	var changes []Change
 
 	for _, c := range r.Changes {
-		if _, ok := seen[c.Module]; !ok {
-			seen[c.Module] = struct{}{}
-			modules = append(modules, c.Module)
+		if c.isWildcard() && strings.HasPrefix(c.Module, prefix) {
+			changes = append(changes, c)
+		}
```

Comment:
I refactored this to make more sense. The function was only ever getting called with a prefix of `""`, so I just modified it so that it returns all wildcards. I think the wildcards are correctly split into core/service client categories.